### PR TITLE
Improve shadow price plotting module

### PR DESCRIPTION
**Pull request by rouille**
### Purpose
Improve `plot_shadowprice_map` module. Changes are mostly related to docstring

### Testing
Ran associated notebook

### Where to look
* `plot_shadowprice_map`:
    + format docstring
    + check type of parameters and raise errors
    + rename variable `hour` to `date`
    + change query for index in data frame and name of column
* `shadowprice_map_demo`: 
    + import `show` from `bokeh.io` and encapsulate the call to `plot_shadowprice`
    + import `pandas` and use `pd.Timestamp` to pass the timestamp of the hour to plot to `plot_shadowprice`

### Usage Example/Visuals
`shadowprice_map_demo` notebook

### Time estimate
10 min

### CODE REVIEW

File: postreise/plot/plot_shadowprice_map.py [hunk 1]
```diff
@@ -297,34 +284,21 @@ def _construct_bus_legend(lmp_split_points):
...
-    :type lmp_split_points: list(float)
-    :param x_range: the x-range for the vbar_stack
-    :type x_range: list(string)
-    :return: bar lengths and labels for the bus legend
-    :rtype: dict, float, dict
+    """Get the bar lengths and labels for the bus legend
+
+    :param list/tuple/set/numpy.array lmp_split_points: lmp values chosen to split
+        the bus data
+    :param list x_range: the x-range for the vertical bar
```
Comment (danielolsen) [inline postreise/plot/plot_shadowprice_map.py:291]: It seems that this parameter is never actually used in this function, and the relevant info could be added to the `bars` dict within `_construct_bus_legend`, after the return from `_get_bus_legend_bars_and_labels`.

File: postreise/plot/plot_shadowprice_map.py [hunk 2]
```diff
@@ -73,28 +77,22 @@ def _get_shadowprice_data(scenario_id):
     return interconnect, bus_map, s.state.get_lmp(), branch_map, cong_abs
 
 
-def _construct_bus_data(bus_map, lmp, user_set_split_points, hour):
+def _construct_bus_data(bus_map, lmp, user_set_split_points, date):
```
[the remark this answers is not in the document]
Comment (rouille) [reply postreise/plot/plot_shadowprice_map.py]: Done

File: postreise/plot/plot_shadowprice_map.py [hunk 3]
```diff
@@ -7,33 +7,39 @@
 from bokeh.transform import linear_cmap
 from powersimdata.scenario.scenario import Scenario
 
+from postreise.analyze.check import _check_date
 from postreise.plot.multi.constants import SHADOW_PRICE_COLORS
```
[the remark this answers is not in the document]
Comment (rouille) [reply postreise/plot/plot_shadowprice_map.py]: Done

File: postreise/plot/plot_shadowprice_map.py (hunk 1 above)
Comment (rouille) [reply postreise/plot/plot_shadowprice_map.py:291]: Yes I saw that. Same thing with `bar_length_sum`, it is equal to `max(bars.values)` and hence does not need to be returned.

Comment (rouille) [reply postreise/plot/plot_shadowprice_map.py]: Done

File: postreise/plot/plot_shadowprice_map.py [hunk 4]
```diff
@@ -270,19 +255,19 @@ def _construct_bus_legend(lmp_split_points):
...
     )
     p.vbar_stack(
-        list(bars.keys())[1:],
+        [*bars],
         x="x_range",
         width=0.9,
-        color=SHADOW_PRICE_COLORS[: (len(bars) - 1)],
-        source=bars,
+        color=shadow_price_pallette[: len(bars)],
+        source={**{"x_range": [""]}, **bars},
```
Comment (danielolsen) [inline postreise/plot/plot_shadowprice_map.py:262]: Clever.

**Review by danielolsen — APPROVED** [review]
Looks good, thanks.

**Pull request merged**